lib/mongo/client.rb

```diff
@@ -665,6 +684,7 @@ def write_concern
     # @since 2.1.0
     def close
       @cluster.disconnect!
+      teardown_encrypter
```

Comment:
Not sure if this is a good way to go about this.

Reply:
Might it make more sense to create and destroy the mongocryptd_client as needed? I know clients are usually intended to be long-running objects, but perhaps this is a reasonable exception? Open to feedback.

Reply:
The purpose of close call is to release resources, for example in a process that will no longer use the client. In light of this, it makes sense to me that close destroys crypt bits as well.

If there isn't already, there should be a test added that close-reconnect sequence produces working crypto.

Reply:
That's a good point! This currently performs no crypto (that's the next couple of tickets), but I will make myself a note to add that as a test case.